src/Function.cpp

```diff
@@ -562,6 +570,43 @@ Function::OutputForwardDecl(std::ostream &out)
 	OutputHeader(out);
 	if(func_attr_inline)
 		out << " __attribute__((always_inline))";
+	if(func_attr_aligned){
+		int probability__BIGGEST_ALIGNMENT__ = 38;
+		bool use__BIGGEST_ALIGNMENT__ = rnd_flipcoin(probability__BIGGEST_ALIGNMENT__);
+		if (use__BIGGEST_ALIGNMENT__)
+			out << " __attribute__((aligned(__BIGGEST_ALIGNMENT__)))";
+		else{
+			int value = 0;
+			int power = rnd_upto(8);
+			if (power == 0)
+				power++;
+			switch (power){
```

Comment:
As already mentioned, please do `a << power` and you don't need a switch statement.
Moreover I would just make a `rnd_upto` a value that will test all powers smaller than `__BIGGEST_ALIGNMENT__`.

Reply:
Removed switch statement. Below is changed code - 

int power = rnd_upto(__BIGGEST_ALIGNMENT__);
                        if (power == 0)
                                power++;
                        value = value << power;

and at last output value.

Reply:
Please push the updated version of the patch to this pull request so that I can comment on the updated version.

Reply:
Sure.